store/sqlstore/status_store.go

```diff
@@ -99,6 +102,89 @@ func (s SqlStatusStore) GetByIds(userIds []string) ([]*model.Status, error) {
 	return statuses, nil
 }
 
+// MySql doesn't has support for RETURNING clause or any euivalent so use transaction
+func (s SqlStatusStore) updateExpiredStatuses(t *gorp.Transaction) ([]*model.Status, error) {
+	selectQuery, selectParams, _ := s.getQueryBuilder().
+		Select("UserId, Status").
+		From("Status").
+		Where(
+			sq.Eq{"Status": model.STATUS_DND},
+			sq.Gt{"DNDEndTimeUnix": -1},
+			sq.LtOrEq{"DNDEndTimeUnix": time.Now().UTC().Unix()},
+		).ToSql()
+	var selectedStatuses []*model.Status
+	_, err := t.Select(&selectedStatuses, selectQuery, selectParams...)
+	if err != nil {
+		return nil, errors.Wrap(err, "getExpiredDNDStatusesT: failed to get expired dnd statuses")
+	}
+	updateQuery, args, _ := s.getQueryBuilder().
+		Update("Status").
+		Where(
+			sq.Eq{"Status": model.STATUS_DND},
+			sq.Gt{"DNDEndTimeUnix": 1},
+			sq.LtOrEq{"DNDEndTimeUnix": time.Now().UTC().Unix()},
+		).
+		Set("Status", "PrevStatus").
```

Comment:
This is setting the status value to the string `"PrevStatus"`, I think you have to use something like `sq.Field` or something like that.